x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.ts

```diff
@@ -4,48 +4,48 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { SearchResponse, SearchHit, SignalHit } from '../../types';
+import { SignalHit } from '../../types';
 import { Logger } from '../../../../../../../../src/core/server';
 import { AlertServices } from '../../../../../alerting/server/types';
+import { SignalSourceHit, SignalSearchResponse, SignalAlertParams } from './types';
 
 // format scroll search result for signals index.
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const buildBulkBody = (doc: SearchHit, signalParams: Record<string, any>): SignalHit => {
-  const indexPatterns = signalParams.index.map((element: string) => `"${element}"`).join(',');
-  const refs = signalParams.references.map((element: string) => `"${element}"`).join(',');
+export const buildBulkBody = (doc: SignalSourceHit, signalParams: SignalAlertParams): SignalHit => {
   return {
     ...doc._source,
     signal: {
+      '@timestamp': new Date().toISOString(),
       rule_revision: 1,
       rule_id: signalParams.id,
       rule_type: signalParams.type,
       parent: {
         id: doc._id,
         type: 'event',
+        index: doc._index,
         depth: 1,
       },
       name: signalParams.name,
       severity: signalParams.severity,
       description: signalParams.description,
-      time_detected: Date.now(),
-      index_patterns: indexPatterns,
-      references: refs,
+      original_time: doc._source['@timestamp'],
+      index_patterns: signalParams.index,
+      references: signalParams.references,
     },
   };
 };
 
 // Bulk Index documents.
 export const singleBulkIndex = async (
-  sr: SearchResponse<object>,
-  params: Record<string, any>, // eslint-disable-line @typescript-eslint/no-explicit-any
+  sr: SignalSearchResponse,
+  params: SignalAlertParams,
   service: AlertServices,
   logger: Logger
 ): Promise<boolean> => {
   if (sr.hits.hits.length === 0) {
     logger.warn('First search result yielded 0 documents');
     return false;
   }
-  const bulkBody = sr.hits.hits.flatMap((doc: SearchHit) => [
+  const bulkBody = sr.hits.hits.flatMap(doc => [
```

Comment:
Should `doc` be typed with `SignalSourceHit`?

Reply:
`doc` is already typed by convention since `SignalSearchResponse` is now more strongly typed.

For example  if you hover over it with your IDE it will show the type of it:
<img width="714" alt="Screen Shot 2019-10-18 at 2 39 01 PM" src="https://user-images.githubusercontent.com/1151048/67126866-55100a80-f1b5-11e9-8fb4-6dce31c28dbe.png">

Pretty much all of the array functions are going to auto carry forward the types unless the array its self is something like an `any`.